blogposts/2022/css-modules-migration.md

```diff
@@ -0,0 +1,210 @@
+3. We have a clear migration path that doesn’t interfere with feature development.
+
+## CSS Modules solution
+
+We researched popular solutions available in the open source universe: trade-offs between CSS in JS, regular CSS, and hybrid solutions. Some offered more powerful features, but we decided to use a solution that we knew we could easily adopt—CSS modules.
+
+### What are CSS Modules?
+
+CSS modules are great because they help you write reusable components with isolated styles. According to the [repo](https://github.com/css-modules/css-modules), CSS modules are:
+
+“CSS files in which all class names and animation names are scoped locally by default.”
+
+_TODO: describe CSS modules application on a simple example with one CSS file and one HTML file._
+
+This approach is designed to fix the problem of the global scope in CSS. Engineers can happily name their CSS selectors whatever they want, without worrying about unintended consequences in other areas of the code. Creating a CSS module is ultimately very similar to creating a typical CSS file. Maintaining this flow ensured we could easily start adopting this approach without interfering with our developer experience too much. After we updated documentation on how to use CSS modules, teams adopted this approach for new features immediately.
+
+## Migration tracking challenges
+
+The Frontend Platform team started the migration by manually converting global styles to CSS modules for a single, recently developed feature. We quickly noticed that despite some initial progress, we had some questions that slowed us down:
+
+1. How can we know whether CSS Modules are continuously used for new features added to the project?
+2. What’s our current progress? What percentage of global styles have already been converted to CSS Modules?
+3. Given the current rate of changes, how much time will it take to migrate all remaining styles to CSS Modules?
+
+We knew we could search the codebase manually for relevant files and make conclusions based on that, or we could write a script to do it for us. But an ideal solution would give a clear, at-a-glance indication of where we were at the moment, and be easy for us to maintain. Thankfully, another product team at Sourcegraph had developed the exact solution to our problem!
+
+“Code Insights reveals high-level information about your codebase, based on both how your code changes over time and its current state.” – [source](https://docs.sourcegraph.com/code_insights).
+
+Code Insights entered Beta in August 2021, and we happily started using it to track the migration progress. As of today, [Code Insights is now Generally Available](LINK TO ANNOUNCEMENT).
+
+<figure>
+  <img src="https://storage.googleapis.com/sourcegraph-assets/blog/code-insights-ga-blogs/migration-to-css-modules-insight.png" alt="Abstract syntax tree">
+  <figcaption>Count of CSS files that are global vs module files.</figcaption>
+</figure>
```

Comment:
TODO: Add link to live insight

Reply:
@umpox @valerybugakov Do you have a link I can use here? We wanted to embed it but that's not possible on the blog yet.

(Also, just noticed the alt text is wrong, I will fix that)

Reply:
A link directly to the code insight? We don't unfortunately, this insight isn't publicly accessible and AFAIK we can't directly link the code insights right now

Reply:
Confirmed with Code Insights team. It's not possible to reference or embed this insight.